src/Abstractions/TaskOrchestrationContext.cs

```diff
@@ -65,6 +65,11 @@ public abstract class TaskOrchestrationContext
     /// </summary>
     public virtual string Version => string.Empty;
 
+    /// <summary>
+    /// Gets the configuration settings for the orchestration context.
+    /// </summary>
+    public virtual IDictionary<string, object?> Properties { get; } = new Dictionary<string, object?>();
```

Comment:
Can this be `IReadOnlyDictionary` instead of `IDictionary`? Seems like there's no scenario where we want customers adding items to this dictionary?